src/X.PagedList/BasePagedList.cs

```diff
@@ -20,16 +20,23 @@ namespace X.PagedList;
 [PublicAPI]
 public abstract class BasePagedList<T> : IPagedList<T>
 {
-    protected readonly List<T> Subset = new();
-
     public const int DefaultPageSize = 100;
 
+    private readonly List<T> _subset = new();
+
     /// <summary>
     /// Parameterless constructor.
     /// </summary>
     protected internal BasePagedList()
     {
     }
+    
+    protected void FillSubset(IEnumerable<T> set)
```

Comment:
I do not like the name. "Fill" does not imply that it is cleared/overwritten.

A better name could be "InitSubset".

To be honest, I do not like the extra layer of complexity this adds at all.

Dealing with the list directly is easier, everybody knows the conventions (in contrast to what a "FullSubset" method does) and it is more flexible.

I would personally drop the entire commit and keep the old version (except for the Obsolete change, which does not belong here).